src/tensorwaves/optimizer/callbacks.py

```diff
@@ -0,0 +1,158 @@
+"""Collection of loggers that can be inserted into an optimizer as callback."""
+
+from abc import ABC, abstractmethod
+from datetime import datetime
+from typing import IO, Iterable, List, Optional
+
+import tensorflow as tf
+import yaml
+from tqdm import tqdm
+
+from tensorwaves.interfaces import Estimator
+
+
+class Callback(ABC):
+    @abstractmethod
+    def __call__(self, parameters: dict, estimator_value: float) -> None:
+        pass
+
+    @abstractmethod
+    def finalize(self) -> None:
+        pass
+
+
+class ProgressBar:
+    def __init__(self) -> None:
+        self.__progress_bar = tqdm()
+
+    def __call__(self, parameters: dict, estimator_value: float) -> None:
+        self.__progress_bar.set_postfix({"estimator": estimator_value})
+        self.__progress_bar.update()
+
+    def finalize(self) -> None:
+        self.__progress_bar.close()
+
+
+class YAMLSummary:
+    def __init__(
+        self,
+        filename: str,
+        estimator: Estimator,
+        step_size: int = 10,
+    ) -> None:
+        """Log fit parameters and the estimator value to a `tf.summary`.
+
+        The logs can be viewed with `TensorBoard
+        <https://www.tensorflow.org/tensorboard>`_ via:
+
+        .. code-block:: bash
+
+            tensorboard --logdir logs
+        """
+        self.__iteration = 0
```

Comment:
`iteration` should be renamed to `function_calls`. Not just in this class. This is just related to the fact that the callbacks in Minuit are called on every function call made. Iteration info could be available in some other optimizers. To be able to handle both (since minuit does not provide this), some design changes would be needed.

Reply:
0a34685